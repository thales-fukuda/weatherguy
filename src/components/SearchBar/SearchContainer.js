import styled from 'styled-components';

const SearchContainer = styled.section`
  position: absolute;
  top: ${props => (props.isWeatherDisplayed ? '25%' : '35%')};
  display: flex;
  flex-direction: column;
  align-items: center;
  transition: 1s ease all;
`;

export default SearchContainer;
