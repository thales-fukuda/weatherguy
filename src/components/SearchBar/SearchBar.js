import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Api from '../../Api';

import Geosuggest from '../elements/Geosuggest';
import Loading from '../elements/Loading';
import Title from '../elements/Title';

import SearchContainer from './SearchContainer';

const SearchBar = (props) => {
  const {
    city,
    weatherStatus,
    updateCity,
    updateWeather,
    updateWeatherStatus,
  } = props;

  const [isLoadingActive, useIsLoadingActive] = useState(false);

  const handleChange = (value) => {
    updateCity(value);
  };

  const handleUpdateWeather = async (value) => {
    try {
      useIsLoadingActive(true);
      const data = await Api.currentWeather(value);
      useIsLoadingActive(false);
      updateWeatherStatus(true);
      updateWeather(data.data);
    } catch (error) {
      console.error(error);
    }
  };

  const handleSuggestSelect = (value) => {
    if (value) {
      handleChange(value.description);
      handleUpdateWeather(value.description);
    }
  };

  const handleKeyPress = (e) => {
    if (e.key === 'Enter') {
      handleUpdateWeather(city);
    }
  };

  return (
    <SearchContainer isWeatherDisplayed={weatherStatus}>
      <Title>Weatherguy</Title>
      <Geosuggest
        types={['(cities)']}
        onChange={e => handleChange(e)}
        onSuggestSelect={value => handleSuggestSelect(value)}
        onKeyDown={e => handleKeyPress(e)}
        placeholder='search for a city...'
        value={city}
      />
      {
        isLoadingActive
        && (
          <Loading>
            <div />
            <div />
            <div />
          </Loading>
        )
      }
    </SearchContainer>
  );
};

SearchBar.propTypes = {
  city: PropTypes.string.isRequired,
  weatherStatus: PropTypes.bool.isRequired,
  updateCity: PropTypes.func.isRequired,
  updateWeather: PropTypes.func.isRequired,
  updateWeatherStatus: PropTypes.func.isRequired,
};

export default SearchBar;
