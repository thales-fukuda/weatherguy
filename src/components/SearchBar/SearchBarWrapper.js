import { connect } from 'react-redux';
import { updateCity, updateWeather } from '../../redux/actions';
import SearchBar from './SearchBar';

const mapStateToProps = state => ({
  city: state.weather.city,
});

const mapDispatchToProps = dispatch => ({
  updateCity: city => dispatch(updateCity(city)),
  updateWeather: city => dispatch(updateWeather(city)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SearchBar);
