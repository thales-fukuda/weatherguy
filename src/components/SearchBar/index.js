import SearchBar from './SearchBarWrapper';

export default SearchBar;
