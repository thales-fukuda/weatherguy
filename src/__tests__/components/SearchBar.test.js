/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SearchBar from '../../components/SearchBar/SearchBar';

describe('<SearchBar/>', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const fakeProps = {
        city: '',
        weatherStatus: true,
        updateCity: () => {},
        updateWeather: () => {},
        updateWeatherStatus: () => {},
      };
      const wrapper = shallow(<SearchBar {...fakeProps} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
