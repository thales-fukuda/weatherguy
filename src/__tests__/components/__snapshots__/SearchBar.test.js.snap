// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar/> render() renders the component 1`] = `
<Fragment>
  <styled.h1
    isWeatherDisplayed={true}
  >
    Weatherguy
  </styled.h1>
  <Styled(default_1$$1)
    isWeatherDisplayed={true}
    onChange={[Function]}
    onKeyDown={[Function]}
    onSuggestSelect={[Function]}
    placeholder="search for a city..."
    types={
      Array [
        "(cities)",
      ]
    }
    value=""
  />
</Fragment>
`;
